### ISSUE TITLE
NW.js Linux export not work

### ISSUE THREAD

**Issue by henrikszucs** [Scirra/Construct-3-bugs#5017]
## Problem description

NW.js Linux export cannot run.

## Attach a .c3p

Any project

## Steps to reproduce

1. Export to nw.js for linux platform
2. Extract
3. Open a folder in terminal
4. Run: sudo ./'Your project' (use your project's name)

## Observed result

Errors in console

## Expected result

Run

## More details
- Tried many nw.js versions but every is broken
- Tried Ubuntu 20.04 LTS and Lubuntu (based in Ubuntu 20.04 LTS)

- The first bug was in xorg display server, I have to install the original Nvidia driver. (This was solved just I mention it) 

- I can run NW.js itself if I download from NW.js' website (https://nwjs.io/downloads/).
- When try to run the exported project console drop errors:
![Képernyőkép – 2021-08-13 12-59-25](https://user-images.githubusercontent.com/47735877/129348299-851bd43e-201e-4e27-880b-d7ac28745842.png)

Solution:
- Construct 3 something encode in compress (Unix systems do not like .zip format).
1. Download and extract project 
2. Download and extract NW.js and merge with project files (replace if same).
3. Now can run project (windows has problem because run in kiosk mode but work) 

## System details

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 92.0.4515.131
Browser engine: Chromium
Context: browser
Operating system: Linux
Operating system version: (unknown)
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 8
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36
C3 release: r259 (beta)
Language setting: en-US

Local storage
Storage quota (approx): 130 gb
Storage usage (approx): 1.6 gb (1.3%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc. (NVIDIA Corporation)
Renderer: ANGLE (NVIDIA Corporation, GeForce GT 640M/PCIe/SSE2, OpenGL 4.5.0 NVIDIA 460.91.03)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 2047
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
Audio information
System sample rate: 44100 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
Are you sure you're extracting the zip on the Linux system? If you do something like extract it first on Windows and then transfer the files to a Linux system, it will lose the file permissions that are necessary for it to be executable. It's only if you extract the zip directly on the Linux system that it will preserve the right file permissions. (Even if you lose the file permissions, it should be possible to fix it by running some terminal commands to set the right file permissions - there's probably an executable file that's missing executable permission.)

**Comment by henrikszucs** [comment]
Yes I do everthing in laptop with native linux operating system. The system details that laptop.

**Comment by henrikszucs** [comment]
I make some test and 2 files are missing:
crashpad_handler
nacl_irt_x86_64.nexe
If I copy that files game are run but without window frame.

**Comment by alastajj** [comment]
It appears to just be a problem with the latest 55 NWJS, because I exported with 54 and 53 and it now works on Linux Mint! 👯‍♂️

**Comment by AshleyScirra** [comment]
Just an update on this: NW.js 0.56.0 was just released and includes the `crashpad_handler` file, but the editor doesn't apply the executable permission to it as it's a new file. The same workaround should work (i.e. copy the file from a download direct from nwjs.io), or you can manually add permissions after export in a Linux terminal with `chmod +rwx crashpad_handler`. The next beta updates the editor to apply the right permissions for this file, so it should work out-of-the-box from there on.

**Comment by henrikszucs** [comment]
In r262 it works. Thank you.
